docs/index.rst

```diff
@@ -221,7 +221,7 @@ To convert a pixel (x, y) = (1, 2) to sky coordinates, call the WCS object as a
 The :meth:`~gwcs.wcs.WCS.invert` method evaluates the :meth:`~gwcs.wcs.WCS.backward_transform`
 if available, otherwise applies an iterative method to calculate the reverse coordinates.
 
-  >>> wcsobj.invert(sky)
+  >>> wcsobj.invert(sky, with_units=True)
```

Comment:
This doc change highlights one of the issues I have encountered today: It seems that `invert` would happily accept high level coordinate objects with `with_units` set to `False`. This seems wrong?

Reply:
This covers the case when the transform does not use units but the input values do, e.g. a SkyCoord object can be the input to a numerical only transform.

Reply:
I am not sure I follow, I handle converting to/from quantity objects depending on the needs of the transform, but what is the desired behaviour of `with_units`? when should it be True and when shouldn't it be?

Reply:
One example is here: https://github.com/spacetelescope/gwcs/pull/457/files#diff-a87a0820dd9de1090bab781695dd821ec8385db4e0aa9f07c98b1f3d972c303dR375-R380

There's separate handling in the API because that has extra requirements to strip / add units as well as what's needed for evaluating the model.

Reply:
> However, if it is it attaches u.pix to the values.

So it should cause the units of the input frame to be attached to the return values?

> This is a change in functionality.

I can probably revert this change, but detecting the input type is a little messy.
Longer term do we want to move to deprecate `with_units` or change its behaviour?

Reply:
The change to the inverse's behaviour for `with_units=True` has been reverted.